src/app/services/data.service.ts

```diff
@@ -412,4 +413,35 @@ export class DataService {
   public getSetupCompleted(): boolean {
     return this.data.setupCompleted;
   }
+
+  public setFavRoute(route: Route): void {
+    // this id will be unique and used as reference
+    // ToThink: What about having favourites A to B and B to A which are the same route but are not?
+    const newId = route.from.place_id + route.to.place_id;
+
+    const favRoute : FavRoute = {
+      id: newId,
+      from: route.from,
+      to: route.to
+    } 
+
+
+    this.data.favRoutes[newId] = favRoute;
+    this.setStorage();
+  }
+
+  public deleteFavRoute(route: Route): void {
+    const deleteId = route.from.place_id + route.to.place_id;
+
+    delete this.data.favRoutes[deleteId];
+    this.setStorage();
+  }
+
+  public getFavRoutes(): { [id: string]: FavRoute} {
+    return this.data.favRoutes;
+  }
+
+  public getFavRoutesArray(): FavRoute[] {
+    return Object.values(this.getFavRoutes());
```

Comment:
i'm getting a typeError: undefind, when loading the search-view. Also i cannot set new fav-routes, since this.data.favRoutes is undefined. 
![favRoutes-Issue](https://user-images.githubusercontent.com/26243342/106392903-3e1a1980-63f4-11eb-9c25-a7a99b2a2b41.png)
Does anyone have the same problem?

Reply:
@Paschild please reset your local storage - thats likely the cause

Reply:
You are right, thanks.